src/main/java/org/cryptomator/ui/mainwindow/VaultListController.java

```diff
@@ -204,6 +226,26 @@ private void selectedVaultDidChange(@SuppressWarnings("unused") ObservableValue<
 		VaultListManager.redetermineVaultState(newValue);
 	}
 
+	private Optional<File> chooseValidVaultDirectory() {
+		DirectoryChooser directoryChooser = new DirectoryChooser();
+		File selectedDirectory;
+
+		do {
+			selectedDirectory = directoryChooser.showDialog(mainWindow);
+			if (selectedDirectory == null) {
+				return Optional.empty();
+			}
+
+			Path selectedPath = selectedDirectory.toPath();
+			if (!Files.isDirectory(selectedPath.resolve(Constants.DATA_DIR_NAME))) {
+				dialogs.prepareNoDDirectorySelectedDialog(mainWindow).build().showAndWait();
+				selectedDirectory = null;
+			}
+		} while (selectedDirectory == null);
+
+		return Optional.of(selectedDirectory);
```

Comment:
If only the path of the selected directory is used an none of the `File` methods, you can directly translate it to a path. THen you don't need to convert it everytime you use it.